docs/database-classes.md

```diff
@@ -1,348 +1,283 @@
 
-For example, this is a valid database class with a constructor:
+For exaαmple, here is our `Person` class with a custom default constructor:
 
 ```csharp
+using System;
 using Starcounter.Database;
 
 [Database]
-public abstract class Order
+public abstract class Person
 {
-    public Order()
+    public abstract string Name { get; set; }
+    public abstract DateTime CreatedAt { get; set; }
+
+    public Person()
     {
-        this.Created = DateTime.Now;
+        CreatedAt = DateTime.UtcNow;
     }
-
-    public abstract DateTime Created { get; set; }
 }
 ```
 
-It's possible to have constructors with parameters, although, they are never called when using `IDatabaseContext.Insert`. Constructors with parameters in database classes can be useful for unit testing purposes when you want to inject dependencies or other arguments into a class. If you add a constructor with parameters to a database class, you also have to add a default constructor.
-
-**Warning**: All C\# access modifiers are accepted for constructors, except for `internal`. Using it will throw `ScErrSchemaCodeMismatch (SCERR4177)` exception.
+It's allowed to have non-default constructors in database class definitions as well, although they are not called by `IDatabaseContext.Insert<T>()`. Instead they can be useful in unit testing scenarios when you want to inject dependencies or other arguments into a class instance. If you add a constructor with parameters to a database class, you also have to add a default constructor.
 
 ## Fields and properties
 
-Database classes should only use properties - either **abstract** or with an explicitly **declared body**. Abstract properties should be public and have a data type that is supported by the database.
-
-Any property that is neither abstract nor defines a body will generate an error.
+Table columns are defined in database classes by abstract instance auto-implemented properties with public get and set accessors and with one of the [supported data types](database-types). The `Person` database class below defines two such columns, `Name` and `CreatedAt`:
 
 ```csharp
-public abstract class Supported
+using System;
+using Starcounter.Database;
+
+[Database]
+public abstract class Person
 {
     public abstract string Name { get; set; }
-    public string CapitalName { get => Name.ToUpper(); }
+    public abstract DateTime CreatedAt { get; set; }
 }
 ```
 
+Database classes can also define calculated properties, which are properties that do not expose an instance field. Commonly, we use them to reflect an aspect of the state of a database class instance. Let's say we want a shorthand for calculating the length of the `Name` string of a `Person`. We could do this by introducing a calculated property. These computed properties do not have the same restrictions as column properties, since they are not bound to the corresponding database table.
```

Comment:
```suggestion
Database classes can also define calculated properties, which are properties that do not expose an instance field. Commonly, we use them to reflect an aspect of the state of a database class instance. Let's say we want a shorthand for calculating the length of the `Name` string of a `Person`. We could do this by introducing a calculated property. These computed properties do not have the same restrictions as column properties, since they are not persistent and are not bound to the corresponding database table.
```